torchao/prototype/safetensors/safetensors_support.py

```diff
@@ -14,40 +13,52 @@
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def load_tensor_state_dict(file_path: str, device: str):
+def unflatten_tensor_state_dict(
+    tensors_data: Dict[str, Any],
+    metadata: Dict[str, Any],
```

Comment:
nit: also the names should be tensor_data_dict and metadata_dict? to reflect the type? please also update the occurrences in docs and tests as well